src/mongocxx/test/versioned_api.cpp

```diff
@@ -0,0 +1,173 @@
+// Copyright 2021 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <bsoncxx/test_util/catch.hh>
+#include <mongocxx/client.hpp>
+#include <mongocxx/exception/operation_exception.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/options/client.hpp>
+#include <mongocxx/options/server_api.hpp>
+#include <mongocxx/test_util/client_helpers.hh>
+#include <mongocxx/uri.hpp>
+
+using namespace mongocxx;
+
+// We'll format many of these examples by hand
+// clang-format off
+
+TEST_CASE("Versioned API, non-strict") {
+    instance::current();
+    // Start Versioned API Example 1
+    using namespace mongocxx;
```

Comment:
note: This appears at global scope earlier in the file-- doesn't hurt anything, and makes it clear to the user that it's part of the example code.